### DISCUSSION TITLE
Include UTXO cborHex with result.

### DISCUSSION THREAD

**Discussion by bakon11** [CardanoSolutions/kupo#83] [Ideas]
#### Describe your idea, in simple words.
Currently using CIP-30 dapp connector when one uses `api.getUtxos()` the result is returned in `cborHex` which you can further feed into the `Cardano Serialization Library` UTXO selector based on CIP-2.

I was wondering how feasible would it be for Kupo to also return the `cborHex` of utxo.
```json
[
  {
    "transaction_id": "35d8340cd6a5d31bf9d09706b92adedf9b1b632e682fdab9fc8865ee3de14e09",
    "output_index": 2,
    "address": "addr1vy3qpx09uscywhpp0ekg9zwmq2yj5vp08husfq6qyh2mpps865j6t",
    "value": {
      "coins": 42,
      "assets": {
        "5d5b08661220099e5e430475c219518179efc7e6c8289db028904834025d5b086": 231,
        "efc7e6c8289db028904834025d5b085d5b08661220099e5e430475c2195181796.08661220099e": 1
      }
    },
    "datum_hash": "309706b92ad8340cd6a5d31bf9d2e682fdab9fc8865ee3de14e09dedf9b1b635",
    "created_at": {
      "slot_no": 51540727,
      "header_hash": "9d09706b92adedf9b1b632e682fdab9fc8865ee3de14e0935d8340cd6a5d31bf"
    },
    "spent_at": {
      "slot_no": 51540727,
      "header_hash": "9d09706b92adedf9b1b632e682fdab9fc8865ee3de14e0935d8340cd6a5d31bf"
    },
    "cborHex": "84a50089825820d9eccdc1a7c28f1a52aa6dc43b14a215db056db10ebc676d16549d00dc15a3d102825820d9eccdc1a7c2" 
  }
]
```
#### Why is it a good idea?
Currently dapp developers using the cip-30 standard dapp connector use `api.getUtxos()` to receive unspent UTXOs for an address, if wanting to use solutions like Kupo to pull address UTXOs to create transactions using CSL, a developer would have to build parsers to process UTXOs into CSL for processing.

If Kupo would return the CBOR of the unspent UTXO they would be able to feed that directly into their dapp CSL colde and make transition between web wallet and local ran Kupo instance seamless.

**Comment by KtorZ** [comment]
Thanks for bringing this up. I had a similar idea to solve this particular problem which I find a little more _"elegant"_ when it comes to building an HTTP API: leverage the content-type. 

Right now, the content-type is `application/json` so you get back JSON, but I was thinking to have the server also support `application/cbor`  to get something along those lines. According to [CIP-0030](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030#transactionunspentoutput) , the expected format for the UTxO is following the [on-chain binary format](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/babbage.cddl). So that is:

``` cddl
transaction_unspent_output = 
    [ input: transaction_input
    , output: transaction_output
    ]

transaction_input = 
    [ transaction_id : $hash32
    , index : uint
    ]

transaction_output = 
      legacy_transaction_output 
    / post_alonzo_transaction_output 

legacy_transaction_output =
    [ address
    , amount : value
    , ? datum_hash : $hash32
    ]

post_alonzo_transaction_output =
    { 0 : address
    , 1 : value
    , ? 2 : datum_option 
    , ? 3 : script_ref   
    }
```

**Comment by bakon11** [comment]
Hi,

Just revisiting this topic again, was wondering if there is still interest in having Kupo also provide the CBOR hex for each respective UTOX in form of CIP30 standard mentioned above.

I started working on a function in JS that will parse kupo query and parse it into CIP30 standard, but am running into some issues with the cbor-x lib I'm using in node JS.
